src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -3028,6 +3028,11 @@ private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byt
             return false;
         }
 
+        private bool readObject(CORINFO_OBJECT_STRUCT_* obj, byte* buffer, int bufferSize, int valueOffset)
+        {
+            throw new NotSupportedException();
```

Comment:
What stops the JIT from calling this? Shouldn't it rather just return false?

Reply:
> What stops the JIT from calling this? Shouldn't it rather just return false?

R2R currently never deals with frozen objects so it will never see `CORINFO_OBJECT_HANDLE` in IR, so this exception is here to validate that (and in other APIs accepting `CORINFO_OBJECT_STRUCT_`)